overlord/snapshotstate/backend/backend.go

```diff
@@ -330,52 +331,28 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 type exportMetadata struct {
 	Format int       `json:"format"`
 	Date   time.Time `json:"date"`
+	Files  []string  `json:"files"`
 }
 
 // Export allows exporting a given snapshot set to the given writer.
-func Export(ctx context.Context, setID uint64, w io.Writer) error {
-	// Open all files first and keep the file descriptors open.
-	//
-	// This ensures that even if a snapshot is removed we can
-	// still serve the data. Hence no need for locking.
-	var snapshotFiles []*os.File
+func Export(snapshotFiles []*os.File, w io.Writer) error {
 	defer func() {
 		for _, f := range snapshotFiles {
```

Comment:
This is now a bit strange, you pass a bunch of open file descriptors in and as a side-effect they are closed after the function was called. So it's impossible to call Export() twice after PrepareExport().

Reply:
I refactored this a bit now